src/components/app.js

```diff
@@ -28,19 +28,34 @@ class App extends Component {
     });
   };
 
-  handleSearch = (e) => {
-    this.setState({search: e.target.value});
+  handleSearchContact = (e) => {
+    this.setState({
+      searchValue: e.target.value,
+    });
+  };
+
+  handleAddContact = (contact) => {
+    const { contactList } = this.state;
+
+    contactList.push(contact);
+
+    this.setState({
+      contactList,
+    });
   };
 
   render() {
-    const { contactList, search } = this.state;
+    const { contactList, searchValue } = this.state;
 
     return (
-      <div className="view_wrap">
-        <h1 className="view_title">Contact List</h1>
-        <SearchBar onSearch={this.handleSearch} search={search} />
-        <ContactList contactList={filterList(contactList, search)} onDeleteItem={this.handleDeleteItem} />
-      </div>
+      <BrowserRouter>
+        <Route exact path="/" render={
+          () => <ContactListPage searchValue={this.state.searchValue} contactList={this.state.contactList} onDeleteContact={this.handleDeleteContact} onSearchContact={this.handleSearchContact} />
```

Comment:
Why you use `this.state` here, you destructured this values above

Reply:
Updated, replaced with destructed value.